wolfssl/wolfcrypt/wc_lms.h

```diff
@@ -19,5 +19,447 @@
+
+/* Maximum number of levels of trees described in private key. */
+#define HSS_MAX_LEVELS              8
+/* Length of full Q in bytes. Q from all levels combined. */
+#define HSS_Q_LEN                   8
+
+/* Compressed parameter set length in bytes. */
+#define HSS_COMPRESS_PARAM_SET_LEN  1
+/* Total compressed parameter set length for private key in bytes. */
+#define HSS_PRIV_KEY_PARAM_SET_LEN  \
+    (HSS_COMPRESS_PARAM_SET_LEN * HSS_MAX_LEVELS)
+
+/* Private key length for one level. */
+#define LMS_PRIV_LEN                \
+    (LMS_Q_LEN + LMS_SEED_LEN + LMS_I_LEN)
+/* Public key length in signature. */
+#define LMS_PUBKEY_LEN              \
+    (LMS_TYPE_LEN + LMS_TYPE_LEN + LMS_I_LEN + LMS_MAX_NODE_LEN)
+
+/* LMS signature data length. */
+#define LMS_SIG_LEN(h, p)                                                   \
+    (LMS_Q_LEN + LMS_TYPE_LEN + LMS_MAX_NODE_LEN + (p) * LMS_MAX_NODE_LEN + \
+     LMS_TYPE_LEN + (h) * LMS_MAX_NODE_LEN)
+
+/* Length of public key. */
+#define HSS_PUBLIC_KEY_LEN          (LMS_L_LEN + LMS_PUBKEY_LEN)
+/* Length of private key. */
+#define HSS_PRIVATE_KEY_LEN         \
+    (HSS_Q_LEN + HSS_PRIV_KEY_PARAM_SET_LEN + LMS_SEED_LEN + LMS_I_LEN)
+/* Maximum public key length - length is constant for all parameters. */
+#define HSS_MAX_PRIVATE_KEY_LEN     HSS_PRIVATE_KEY_LEN
+/* Maximum private key length - length is constant for all parameters. */
+#define HSS_MAX_PUBLIC_KEY_LEN      HSS_PUBLIC_KEY_LEN
+/* Maximum signature length. */
+#define HSS_MAX_SIG_LEN                                                        \
+    (LMS_TYPE_LEN +                                                            \
+     LMS_MAX_LEVELS * (LMS_Q_LEN + LMS_TYPE_LEN + LMS_TYPE_LEN +               \
+                       LMS_MAX_NODE_LEN * (1 + LMS_MAX_P + LMS_MAX_HEIGHT)) +  \
+     (LMS_MAX_LEVELS - 1) * LMS_PUBKEY_LEN                                     \
+     )
+
+/* Maximum buffer length required for use when hashing. */
+#define LMS_MAX_BUFFER_LEN          \
+    (LMS_I_LEN + LMS_Q_LEN + LMS_P_LEN + LMS_W_LEN + 2 * LMS_MAX_NODE_LEN)
+
+
+/* Private key data length.
+ *
+ * HSSPrivKey.priv
+ */
+#define LMS_PRIV_KEY_LEN(l) \
+    ((l) * LMS_PRIV_LEN)
+
+/* Stack of nodes. */
+#define LMS_STACK_CACHE_LEN(h)              \
+     (((h) + 1) * LMS_MAX_NODE_LEN)
+
+/* Root cache length. */
+#define LMS_ROOT_CACHE_LEN(rl)              \
+    (((1 << (rl)) - 1) * LMS_MAX_NODE_LEN)
+
+/* Leaf cache length. */
+#define LMS_LEAF_CACHE_LEN(cb)              \
+    ((1 << (cb)) * LMS_MAX_NODE_LEN)
+
+/* Length of LMS private key state.
+ *
+ * LmsPrivState
+ *   auth_path +
+ *   root +
+ *   stack.stack + stack.offset +
+ *   cache.leaf + cache.index + cache.offset
+ */
+#define LMS_PRIV_STATE_LEN(h, rl, cb)   \
+    (((h) * LMS_MAX_NODE_LEN) +         \
+     LMS_STACK_CACHE_LEN(h) + 4 +       \
+     LMS_ROOT_CACHE_LEN(rl) +           \
+     LMS_LEAF_CACHE_LEN(cb) + 4 + 4)
+
+#ifndef WOLFSSL_LMS_SMALL
```

Comment:
`WOLFSSL_LMS_SMALL` here and other places.

I think this is missing changes from recent PRs.

Reply:
fixed